src/Ng/Program.cs

```diff
@@ -92,7 +93,7 @@ public static async Task MainAsync(string[] args)
                 job = NgJobFactory.GetJob(jobName, telemetryService, loggerFactory);
                 await job.RunAsync(arguments, cancellationTokenSource.Token);
             }
-            catch (ArgumentException ae)
+            catch (Exception ae) when (ae is ArgumentException || ae is KeyNotFoundException)
             {
                 _logger?.LogError("A required argument was not found or was malformed/invalid: {Exception}", ae);
```

Comment:
Maybe improve the log message to denote that the key is not found, or another catch block for this case.

Reply:
Good idea, I'll do that.